manager/controlapi/task.go

```diff
@@ -123,6 +125,14 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 				return filterContains(e.NodeID, request.Filters.NodeIDs)
 			},
 			func(e *api.Task) bool {
+				r, err := naming.Runtime(e.Spec)
```

Comment:
Very minor: I would try to avoid this call to `Runtime` when no runtime filter is specified. Could we have something above like:

```go
if len(request.Filters.Runtimes) == 0 {
        return true
}
```

and then the `return` below can be simplified to `return false`.